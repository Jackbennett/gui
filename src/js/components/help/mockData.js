export const helpProps = {
  docsVersion: '',
  findLocalIpAddress: jest.fn,
  isHosted: false,
  isEnterprise: false,
  menderDebPackageVersion: 'master',
  menderVersion: 'master',
  menderArtifactVersion: 'master',
  org: {
    tenant_token: ''
  }
};
