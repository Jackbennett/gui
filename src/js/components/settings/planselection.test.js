import React from 'react';
import { render } from '@testing-library/react';
import PlanSelection from './planselection';
import { undefineds } from '../../../../tests/mockData';

describe('PlanSelection component', () => {
  it(`renders correctly`, () => {
    const { baseElement } = render(<PlanSelection isUpgrade offerValid offerTag="very now" trial updatedPlan="enterprise" />);
    const view = baseElement.firstChild;
    expect(view).toMatchSnapshot();
    expect(view).toEqual(expect.not.stringMatching(undefineds));
  });
});
