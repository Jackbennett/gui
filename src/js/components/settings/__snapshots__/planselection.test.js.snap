// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlanSelection component renders correctly 1`] = `
<div
  class="flexbox space-between"
  style="padding-bottom: 15px;"
>
  <div
    class="planPanel "
  >
    <div
      class="uppercased align-center muted"
      style="margin-bottom: -11px; font-size: smaller;"
    >
      current plan
    </div>
    <h4>
      Starter
       
      very now
    </h4>
    <div>
      <div
        class="link-color bold"
        style="font-size: 1rem;"
      >
        $23/month for first 50 devices
      </div>
      <div
        class="pre-line"
      >
        for first 6 months;
$29/month thereafter
      </div>
    </div>
    <ul
      class="unstyled"
    >
      <li>
        <span
          class="info"
        >
          Basic OTA features
        </span>
      </li>
    </ul>
  </div>
  <div
    class="planPanel "
  >
    <div
      class="uppercased align-center muted"
      style="margin-bottom: -11px; font-size: smaller;"
    >
      upgrade
    </div>
    <h4>
      Professional
       
      very now
    </h4>
    <div>
      <div
        class="link-color bold"
        style="font-size: 1rem;"
      >
        $200/month for first 50 devices
      </div>
      <div
        class="pre-line"
      >
        for first 6 months;
$249/month thereafter
      </div>
    </div>
    <ul
      class="unstyled"
    >
      <li>
        <span
          class="info"
        >
          + Advanced OTA features
        </span>
      </li>
      <li>
        <span
          class="info"
        >
          + Standard support
        </span>
      </li>
    </ul>
  </div>
  <div
    class="planPanel active"
  >
    <div
      class="uppercased align-center muted"
      style="margin-bottom: -11px; font-size: smaller;"
    >
      upgrade
    </div>
    <h4>
      Enterprise
       
    </h4>
    <div>
      <div
        class="link-color bold"
        style="font-size: 1rem;"
      >
        Custom pricing
      </div>
      <div>
        unlimited devices
      </div>
    </div>
    <ul
      class="unstyled"
    >
      <li>
        <span
          class="info"
        >
          + Enterprise-grade OTA features
        </span>
      </li>
      <li>
        <span
          class="info"
        >
          + Premium support
        </span>
      </li>
    </ul>
  </div>
</div>
`;
